### ISSUE TITLE
replacing std::regex with ctre 

### ISSUE THREAD

**Issue by andijcr** [docopt/docopt.cpp#141]
I replaced std::regex with https://github.com/hanickadot/compile-time-regular-expressions .
In doing so I had to add a .clang-format, remove boost regex, and move the standard to c++17 (this last one is a hard requirement for ctre).
Is there an interest in these changes, or is it out of scope for this project?

you can find my fork here:
https://github.com/andijcr/docopt.cpp/tree/feat/ctre

**Comment by nfrmtkr** [comment]
I would vote for it because std::regex of Microsoft Visual C++ has mayn issues. The one you'll hit very soon is:
[Grouping within repetition causes regex stack error](https://developercommunity.visualstudio.com/t/grouping-within-repetition-causes-regex-stack-erro/885115)
However, I would prefer a plugable regex approach allowing to add your own regex library. For instance PCRE.

**Comment by jaredgrubb** [comment]
I do like this idea ... does it all work ok with CTRE? Whats the compile time change?

**Comment by andijcr** [comment]
I opened a [pull request](https://github.com/docopt/docopt.cpp/pull/143) to compare the changes. The important changes are in this commit https://github.com/andijcr/docopt.cpp/commit/ca98f8496e0728371af5b0ae8a3fb26f1d3e4892 
The compile time tends to be worse than before, but it's limited only to a translation unit of the library, so it's a price paid only once when at the first build.
The major problem it's the hard requirement for c++20, but since the changes are localized I think some ifdefs should be able to handle it.
